interpreter/backend/compiler.py

```
@@ -1,300 +1,455 @@
+        """
+        for i, existing in enumerate(self.constants):
+            if type(existing) is type(value):
+                if hasattr(value, "value") and existing.value == value.value:
                     return i
+            elif existing == value:
+                return i
         self.constants.append(value)
         return len(self.constants) - 1
 
-    def compile(self, node):
+    def compile(self, node: Node | None) -> None | ProgramArtifacts:
```

Ruff (0.16.2)

[warning] `None` not at the end of the type union.

Move `None` to the end of the type union

(RUF036)